src/shared/api/schemas.ts

```diff
@@ -0,0 +1,308 @@
+  page: z.number(),
+  per_page: z.number(),
+  total_pages: z.number(),
+  total_items: z.number(),
+});
+
+export type Pagination = z.infer<typeof paginationSchema>;
+
+export const paginatedSchema = <T extends z.ZodType>(item: T) =>
+  z.object({
+    pagination: paginationSchema,
+    items: item.array(),
+  });
+
+export interface Paginated<T> {
+  pagination: Pagination
+  items: T[]
+}
+
+const paginationParamsSchema = z
+  .object({
+    page: z.number(),
+    per_page: z.number(),
+  })
+  .partial();
+
+export type PaginationParams = z.infer<typeof paginationParamsSchema>;
+
+const metadataSchema = z.record(z.string(), z.any());
+
+export class DomainId {
+  private domain: string;
+  constructor(domain: string) {
+    this.domain = domain;
+  }
+  public getDomain() {
+    return this.domain;
+  }
+}
+
+export class AccountId {
+  private signatory: string;
+  private accountDomain: string;
+  public constructor(signatory: string, domain: string) {
+    this.signatory = signatory;
+    this.accountDomain = domain;
+  }
+
+  public getSignatory() {
+    return this.signatory;
+  }
+  public getAccountDomain() {
+    return this.accountDomain;
+  }
+  public toString() {
+    return this.signatory + '@' + this.accountDomain;
+  }
+}
+
+export function transformToAccountId(accountId: string): AccountId {
+  const [account, account_domain] = accountId.split('@');
+
+  return new AccountId(account, account_domain);
+}
+
+export class AssetDefinitionId {
+  private name: string;
+  private domain: DomainId;
+
+  public constructor(name: string, domain: string) {
+    this.name = name;
+    this.domain = new DomainId(domain);
+  }
+
+  public getName() {
+    return this.name;
+  }
+  public getDomain() {
+    return this.domain.getDomain();
+  }
+  public toString() {
+    return this.name + '#' + this.domain.getDomain();
+  }
+}
+
+export function transformToAssetDefinitionId(assetDefinitionId: string): AssetDefinitionId {
+  const [asset, asset_domain] = assetDefinitionId.split('#');
+
+  return new AssetDefinitionId(asset, asset_domain);
+}
+
+export class AssetId {
+  private account: AccountId;
+  private definition: AssetDefinitionId;
+
+  public constructor(data: { asset: string, asset_domain: string, signatory: string, account_domain: string }) {
+    this.account = new AccountId(data.signatory, data.account_domain);
+    this.definition = new AssetDefinitionId(data.asset, data.asset_domain ? data.asset_domain : data.account_domain);
+  }
+
+  public get asset() {
+    return this.definition.getName();
+  }
+  public get domain() {
+    return this.definition.getDomain();
+  }
+  public get assetDefinition() {
+    return this.definition.toString();
+  }
+  public get signatory() {
+    return this.account.getSignatory();
+  }
+  public get accountDomain() {
+    return this.account.getAccountDomain();
+  }
+  public toString() {
+    return this.definition.toString() + '#' + this.account.toString();
```

Comment:
The logic is a bit different.

If the definition's domain equals to the account's domain, the ID is shortened. Please refer to the API docs where I gave an example for `AssetId` model.